viper.go

```diff
@@ -761,7 +764,12 @@ func (v *Viper) find(key string) interface{} {
 			return cast.ToBool(flag.ValueString())
 		case "stringSlice":
 			s := strings.TrimPrefix(flag.ValueString(), "[")
-			return strings.TrimSuffix(s, "]")
+			s = strings.TrimSuffix(s, "]")
+			if len(s) == 0 {
+				return []string{}
+			}
+			res, _ := readAsCSV(s)
```

Comment:
Is the call to `readAsCSV` necessary?

I assume you'd use it instead of string split to to handle this case: `go run main.go -t 'one,"two three"'`

But when I run this command on my old version of viper (7fb2782) I get this:
```text
(string) (len=15) "[one,two three]"
([]string) (len=2 cap=2) {
 (string) (len=8) "[one,two",
 (string) (len=6) "three]"
}
([]string) (len=2 cap=2) {
 (string) (len=3) "one",
 (string) (len=9) "two three"
}
```

Which means pflags is already doing this. Is there no way we can get the correct value out of pflags directly?

This is just a question, like the previous comment. Thanks for taking the time to work on this!